grovedb/src/batch/key_info.rs

```diff
@@ -0,0 +1,107 @@
+use std::{
+    cmp::Ordering,
+    hash::{Hash, Hasher},
+};
+
+use storage::worst_case_costs::WorstKeyLength;
+use visualize::{Drawer, Visualize};
+
+use crate::batch::key_info::KeyInfo::{KnownKey, MaxKeySize};
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum KeyInfo {
+    KnownKey(Vec<u8>),
+    MaxKeySize { unique_id: Vec<u8>, max_size: u8 },
+}
+
+impl PartialOrd<Self> for KeyInfo {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        match self.as_slice().partial_cmp(other.as_slice()) {
+            None => None,
+            Some(ord) => match ord {
+                Ordering::Less => Some(Ordering::Less),
+                Ordering::Equal => {
+                    let other_len = other.len();
+                    match self.len().partial_cmp(&other_len) {
+                        None => Some(Ordering::Equal),
+                        Some(ord) => Some(ord),
+                    }
+                }
+                Ordering::Greater => Some(Ordering::Less),
+            },
+        }
+    }
+}
+
+impl Ord for KeyInfo {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.as_slice().cmp(other.as_slice())
```

Comment:
Viewing at PR once again for some reason my comment is missed here so I'll point this out once again: `PartialOrd` and `Ord` for `KeyInfo` use different implementation.

`Implementations must be consistent with the PartialOrd implementation, and ensure max, min, and clamp are consistent with cmp:` (https://doc.rust-lang.org/stable/std/cmp/trait.Ord.html)